test/others/criu-ns/run.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+
+import os
+import pty
+import shutil
+import subprocess
+import sys
+import time
+
+
+CRIU_BIN = "../../../criu/criu"
+CRIU_NS = "../../../scripts/criu-ns"
+
+
+def check_dumpdir():
+    if os.path.isdir("dumpdir"):
+        shutil.rmtree("dumpdir")
+    os.mkdir("dumpdir", 0o755)
+
+
+def create_pty():
+    fd_m, fd_s = pty.openpty()
+    return (os.fdopen(fd_m, "wb"), os.fdopen(fd_s, "wb"))
+
+
+def test_dump_and_restore_with_shell_job():
+    check_dumpdir()
+
+    open("running", "w").close()
+    create_pty()
+    pid = os.fork()
+    if pid == 0:
+        os.setsid()
```

Comment:
--shell-job options fixes sessions and terminals

If we do setsid here we test only part of what --shell-job does, we need to remove it in child.

When you remove it, you would need to add os.setsid() in the begining of main to fix namespace related problems (sid from upper level of pidns is not supported by criu).

Reply:
I agree. Initially,  I came to the same conclusion as well. But, when executing `make -C test/others/criu-ns/ run` without `os.setsid()` the `test_dump_and_restore_with_shell_job()` always failed. So, that's why I kept it here (in child). Moving the `os.setsid()` from child to the main fixes that. Thanks!

Reply:
Please don't forget to move it in your PR. =)

Reply:
Done, thanks!